### PR TITLE
erdpy proxy improvement

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Add the functionality of sending a transaction, and waiting for the result.

### CODE REVIEW

File: erdpy/proxy/core.py [hunk 1]
```diff
@@ -120,3 +120,16 @@ def get_hyperblock(self, key) -> Any:
         response = do_get(url)
         response = response.get("hyperblock", {})
         return response
+
+    def send_transaction_and_wait_for_result(self, payload: Any) -> str:
+        url = f"{self.url}/transaction/send"
+        response = do_post(url, payload)
+        tx_hash = response.get("txHash")
+
+        while True:
```
Comment (iulianpascalau) [inline erdpy/proxy/core.py:129]: Although this works, if we send a wrong transaction the whole script will block. Maybe add a timeout value in seconds? and make this function return both str and error?

Comment (BeniaminDrasovean) [reply erdpy/proxy/core.py]: Done.

File: erdpy/proxy/core.py [hunk 2]
```diff
@@ -120,3 +121,19 @@ def get_hyperblock(self, key) -> Any:
...
+        num_runs = 100
+        for _ in range(0, num_runs):
+            last_nonce = self.get_last_block_nonce("metachain")
+            last_hyperblock = self.get_hyperblock(last_nonce)
+            finalized_transactions = last_hyperblock["transactions"]
+            for transaction in finalized_transactions:
+                if transaction["hash"] == tx_hash:
+                    return self.get_transaction(tx_hash=tx_hash, with_results=True)
+            time.sleep(1)
+        return ""
```
Comment (camilbancioiu) [inline erdpy/proxy/core.py:139]:
```suggestion
        for _ in range(0, num_runs):
            if self.is_transaction_finalized(tx_hash):
                return self.get_transaction(tx_hash=tx_hash, with_results=True)
            time.sleep(1)
        return ""
        
   def is_transaction_finalized(self, tx_hash):
        last_nonce = self.get_last_block_nonce("metachain")
        last_hyperblock = self.get_hyperblock(last_nonce)
        finalized_transactions = last_hyperblock["transactions"]
        
        for transaction in finalized_transactions:
                if transaction["hash"] == tx_hash:
                    return True
                    
        return False
```

**Review by camilbancioiu — CHANGES_REQUESTED** [review]
Minor code refactoring, as suggested in comment.

**Comment by camilbancioiu** [comment]
The new function `send_transaction_and_wait_for_result()` doesn't seem to be called yet, is that intentional?

File: erdpy/proxy/core.py (hunk 2 above)
Comment (BeniaminDrasovean) [reply erdpy/proxy/core.py]: Refactored. This will be called from another repo, PR is on the way.

**Review by camilbancioiu — APPROVED** [review]

File: erdpy/proxy/core.py [hunk 3]
```diff
@@ -121,15 +122,24 @@ def get_hyperblock(self, key) -> Any:
...
-        while True:
-            last_nonce = self.get_last_block_nonce("metachain")
-            last_hyperblock = self.get_hyperblock(last_nonce)
-            finalized_transactions = last_hyperblock["transactions"]
-            for transaction in finalized_transactions:
-                if transaction["hash"] == tx_hash:
-                    return self.get_transaction(tx_hash=tx_hash, with_results=True)
+        for _ in range(0, num_seconds_timeout):
+            if self.is_transaction_finalized(tx_hash):
+                return self.get_transaction(tx_hash=tx_hash, with_results=True)
```
Comment (iulianpascalau) [inline erdpy/proxy/core.py:132]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**